src/main/java/stirling/software/SPDF/controller/api/security/WatermarkController.java

```diff
@@ -190,8 +190,14 @@ private void addTextWatermark(
         int watermarkRows = (int) (pageHeight / watermarkHeight + 1);
         int watermarkCols = (int) (pageWidth / watermarkWidth + 1);
 
+        // Calculating the effective line length according to the angle used
+        float effectiveHeight = watermarkHeight / (float) Math.cos(Math.toRadians(rotation));
```

Comment:
Can't this do a /0 if angle is 90 or 270?

Reply:
You are right, I hadn't thought of that. Although Math.cos() never exactly gives zero, it would never throw that exception, but it's risky to keep code like that. I'll think of another way to solve this. Do you have any idea?